diff --git a/src/index.tsx b/src/index.tsx
index c224c6c..7fa6589 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -1,7 +1,7 @@
 import React from "react"
 export {default as useTemplate} from "./useTemplate"
 export {default as theme} from "./services/theme"
-export {default as i18n} from "./services/i18n"
+export {default as i18n, getLanguages, initializeLanguages} from "./services/i18n"
 export {default as Header} from "./components/Header/Header"
 export {default as Dialog} from "./components/Dialog/Dialog"
 export {default as CustomDropFile} from "./components/CustomDropFile/CustomDropFile"
diff --git a/src/services/i18n.ts b/src/services/i18n.ts
index 51325e7..5e43b51 100644
--- a/src/services/i18n.ts
+++ b/src/services/i18n.ts
@@ -1,33 +1,39 @@
 // SPDX-FileCopyrightText: 2022 Félix Robles <felix@sequentech.io>
 //
 // SPDX-License-Identifier: AGPL-3.0-only
-import i18n, {i18n as I18N} from "i18next"
+import i18n, {i18n as I18N, Resource} from "i18next"
+import {deepmerge} from "@mui/utils"
 import LanguageDetector from "i18next-browser-languagedetector"
 import {initReactI18next} from "react-i18next"
 import englishTranslation from "../translations/en"
 import spanishTranslation from "../translations/es"
 
-i18n.use(LanguageDetector)
-    .use(initReactI18next)
-    .init({
-        // we init with resources
-        resources: {
-            en: englishTranslation,
-            es: spanishTranslation,
-        },
-        fallbackLng: "es",
-        debug: true,
+export const initializeLanguages = (externalTranslations: Resource) => {
+    const libTranslations: Resource = {
+        en: englishTranslation,
+        es: spanishTranslation,
+    }
+    const mergedTranslations = deepmerge(libTranslations, externalTranslations)
 
-        // have a common namespace used around the full app
-        ns: ["translations"],
-        defaultNS: "translations",
+    i18n.use(LanguageDetector)
+        .use(initReactI18next)
+        .init({
+            // we init with resources
+            resources: mergedTranslations,
+            fallbackLng: "es",
+            debug: true,
 
-        keySeparator: ".",
+            // have a common namespace used around the full app
+            ns: ["translations"],
+            defaultNS: "translations",
 
-        interpolation: {
-            escapeValue: false,
-        },
-    })
+            keySeparator: ".",
+
+            interpolation: {
+                escapeValue: false,
+            },
+        })
+}
 
 export const getLanguages = (i18n: I18N) => Object.keys(i18n.services.resourceStore.data)
 
