diff --git a/packages/sequent-core/src/util/convert_vec.rs b/packages/sequent-core/src/util/convert_vec.rs
index 901f5248..2cc302a6 100644
--- a/packages/sequent-core/src/util/convert_vec.rs
+++ b/packages/sequent-core/src/util/convert_vec.rs
@@ -45,6 +45,16 @@ pub fn convert_map(
 ) -> HashMap<String, Vec<String>> {
     original_map
         .into_iter()
-        .map(|(key, value)| (key, value.into_vec()))
+        .map(|(key, value)| {
+            let vec = match value {
+                Value::Array(arr) => arr
+                    .into_iter()
+                    .filter_map(|v| v.as_str().map(|s| s.to_string()))
+                    .collect(),
+                Value::String(s) => vec![s],
+                _ => Vec::new(),
+            };
+            (key, vec)
+        })
         .collect()
-}
+}
\ No newline at end of file
diff --git a/packages/windmill/src/services/users.rs b/packages/windmill/src/services/users.rs
index 3d220f10..2e900b19 100644
--- a/packages/windmill/src/services/users.rs
+++ b/packages/windmill/src/services/users.rs
@@ -288,15 +288,25 @@ pub async fn list_users(
             u.realm_id,
             u.username,
             u.created_timestamp,
-            COALESCE(json_object_agg(attr.name, attr.value) FILTER (WHERE attr.name IS NOT NULL), '{{}}'::json) AS attributes,
+            COALESCE(attr_json.attributes, '{{}}'::json) AS attributes,
             COUNT(u.id) OVER() AS total_count
         FROM
             user_entity AS u
         INNER JOIN
             realm AS ra ON ra.id = u.realm_id
         {area_ids_join_clause}
-        LEFT JOIN
-            user_attribute AS attr ON u.id = attr.user_id
+        LEFT JOIN LATERAL (
+            SELECT
+                json_object_agg(attr.name, attr.values_array) AS attributes
+            FROM (
+                SELECT
+                    ua.name,
+                    json_agg(ua.value) AS values_array
+                FROM user_attribute ua
+                WHERE ua.user_id = u.id
+                GROUP BY ua.name
+            ) attr
+        ) attr_json ON true
         WHERE
             ra.name = $1 AND
             ($4::VARCHAR IS NULL OR email ILIKE $4) AND
@@ -308,8 +318,6 @@ pub async fn list_users(
             {enabled_condition}
             {email_verified_condition}
            AND ({dynamic_attr_clause})
-        GROUP BY
-            u.id
         ORDER BY {sort_clause}
         LIMIT $2 OFFSET $3;
     "#).as_str()).await?;
